regression-tests/sparktkregtests/testcases/frames/frame_create_from_csv_test.py

```diff
@@ -0,0 +1,128 @@
+
+import sys
+import unittest
+from sparktkregtests.lib import sparktk_test
+import sparktk
+
+
+class FrameImportCSVTest(sparktk_test.SparkTKTestCase):
+
+    def setUp(self):
+        """Build frames to be exercised and establish known baselines"""
+        super(FrameImportCSVTest, self).setUp()
+        self.dataset = self.get_file("int_str_int.csv")
+        self.schema = [("num1", int), ("letter", str), ("num2", int)]
+        self.frame = self.context.frame.import_csv(self.dataset, schema=self.schema)
+
+    def test_frame_invalid_column(self):
+        """Tests retrieving an invalid column errors."""
+        with self.assertRaises(ValueError):
+            self.frame.take(100, columns=['not_in'])
+
+    def test_frame_create_row_count(self):
+        """ Trivial Frame creation. """
+        frame = self.context.frame.import_csv(self.dataset, schema=self.schema)
+        self.assertEqual(frame.row_count, 3)
+        self.assertEqual(len(frame.take(3).data), 3)
+
+    def test_schema_duplicate_names_diff_type(self):
+        """CsvFile creation fails with duplicate names, different type."""
+        # double num1's same type
+        bad = [("num1", int), ("num1", str), ("num2", int)]
+        with self.assertRaisesRegexp(Exception, "cannot be more than one character"):
+            self.context.frame.import_csv(self.dataset, bad)
+
+    def test_schema_duplicate_names_same_type(self):
+        """CsvFile creation fails with duplicate names, same type."""
+        # two num1's with same type
+        bad = [("num1", int), ("num1", int), ("num2", int)]
+        with self.assertRaisesRegexp(Exception, "cannot be more than one character"):
+            self.context.frame.import_csv(self.dataset, bad)
+
+    def test_schema_invalid_type(self):
+        """CsvFile cration with a schema of invalid type fails."""
+        bad_schema = -77
+        with self.assertRaisesRegexp(Exception, "cannot be more than one character"):
+            self.context.frame.import_csv(self.dataset, bad_schema)
+
+    def test_schema_invalid_format(self):
+        """CsvFile creation fails with a malformed schema."""
+        bad_schema = [int, int, float, float, str]
+        with self.assertRaisesRegexp(Exception, "cannot be more than one character"):
+            self.context.frame.import_csv(self.dataset, bad_schema)
+
+    def test_frame_delim_colon(self):
+        """Test building a frame with a colon delimiter."""
+        dataset_passwd = self.get_file("passwd.csv")
+        passwd_schema = [("userid", str),
+                         ("password", str),
+                         ("usrNum", int),
+                         ("grpNum", int),
+                         ("userName", str),
+                         ("home", str),
+                         ("shell", str)]
+        passwd_frame = self.context.frame.import_csv(dataset_passwd, schema=passwd_schema, delimiter=':')
+        self.assertEqual(len(passwd_frame.take(1).data[0]), len(passwd_schema))
```

Comment:
Need to compare data while we're at it.  Comparing len is not getting much coverage.